src/safeposix/syscalls/net_calls.rs

```diff
@@ -733,36 +892,51 @@ impl Cage {
 
         let (pipe1, pipe2) = create_unix_sockpipes();
 
+        //Setup the socket handle with the remote address
         sockhandle.remoteaddr = Some(remoteaddr.clone());
         sockhandle.unix_info.as_mut().unwrap().sendpipe = Some(pipe1.clone());
         sockhandle.unix_info.as_mut().unwrap().receivepipe = Some(pipe2.clone());
 
+        //Check if the socket is set to blocking mode
+        //connvar is necessary to synchronize connect and accept
+        //as we are performing it in the user space
         let connvar = if sockfdobj.flags & O_NONBLOCK == 0 {
             Some(interface::RustRfc::new(ConnCondVar::new()))
         } else {
             None
         };
 
-        // receive_pipe and send_pipe need to be swapped here
-        // because the receive_pipe and send_pipe are opposites between the
-        // sender and receiver. Swapping here also means we do not need to swap in
-        // accept.
+        //receive_pipe and send_pipe need to be swapped here because the receive_pipe 
+        //and send_pipe are opposites between the sender and receiver. 
+        //Swapping here also means we do not need to swap in accept.
         let entry = DomsockTableEntry {
             sockaddr: sockhandle.localaddr.unwrap().clone(),
             receive_pipe: Some(pipe1.clone()).unwrap(),
             send_pipe: Some(pipe2.clone()).unwrap(),
             cond_var: connvar.clone(),
         };
+        //Access the domsock_accept_table, which keeps track of socket paths and 
+        //details pertaining to them: the socket address, receive and send pipes, and cond_var
         NET_METADATA
             .domsock_accept_table
             .insert(remotepathbuf, entry);
+        //Update the sock handle state to indicate that it is connected
```

Comment:
Add a BUG or TODO here: there is supposed to be some logics handling nonblocking connects here, which is currently missing